File: Two-Pointer/1019-squares-of-a-sorted-array/squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int l = 0, pos = nums.size(), r = nums.size() - 1;
        vector<int> ans(pos--);

        while(l <= r){
            int a = nums[l] * nums[l];
            int b = nums[r] * nums[r];

            if(a > b){
                ans[pos--] = a;
                l++;
            }
            else{
                ans[pos--] = b;
                r--;
            }
        }

        return ans;
    }
};

File: Two_Pointer/1019-squares-of-a-sorted-array/squares-of-a-sorted-array.py
from typing import List

class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        l, r = 0, len(nums) - 1
        ans = []

        while l <= r:
            a = nums[l] ** 2
            b = nums[r] ** 2

            if a > b:
                ans.append(a)
                l += 1
            else:
                ans.append(b)
                r -= 1

        return ans[::-1]
